src/main/java/de/ait/secondlife/services/OfferServiceImpl.java

```diff
@@ -0,0 +1,209 @@
+package de.ait.secondlife.services;
+
+import de.ait.secondlife.domain.dto.OfferCreationDto;
+import de.ait.secondlife.domain.dto.OfferResponseDto;
+import de.ait.secondlife.domain.dto.OfferResponseWithPaginationDto;
+import de.ait.secondlife.domain.dto.OfferUpdateDto;
+import de.ait.secondlife.domain.entity.Offer;
+import de.ait.secondlife.exception_handling.exeptions.*;
+import de.ait.secondlife.repositories.OfferRepository;
+import de.ait.secondlife.services.interfaces.OfferService;
+import de.ait.secondlife.services.interfaces.StatusSevice;
+import de.ait.secondlife.services.mapping.OfferMappingService;
+import de.ait.secondlife.services.validator.EntityValidator;
+import jakarta.transaction.Transactional;
+import jakarta.validation.ConstraintViolationException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static de.ait.secondlife.constans.StatusConstans.DRAFT_STATUS;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class OfferServiceImpl implements OfferService {
+
+    private final OfferRepository offerRepository;
+    private final OfferMappingService mappingService;
+    private final StatusSevice statusSevice;
+    private final EntityValidator validator;
+
+
+    @Override
+    public OfferResponseWithPaginationDto findOffers(Pageable pageable) {
+        if (pageable == null) throw new PageableIsNullException();
+
+        Page<Offer> pageOfOffer = offerRepository.findAllByIsActiveTrue(pageable);
+
+        return offersToOfferRequestWithPaginationDto(pageOfOffer);
+    }
+
+
+    @Override
+    public OfferResponseDto findOfferById(UUID id) {
+        if (id == null) throw new IdIsNullException();
+
+        Offer offer = offerRepository.findByIdAndIsActiveTrue(id)
+                .orElseThrow(() -> new OfferNotFoundException(id));
+
+        try {
+            return mappingService.toRequestDto(offer);
+        } catch (Exception e) {
+            throw new MappingException(offer, e.getMessage());
+        }
+    }
+
+    @Override
+    public OfferResponseWithPaginationDto findOffersByUserId(Long id, Pageable pageable) {
+        if (id == null) throw new IdIsNullException();
+        Page<Offer> pageOfOffer = offerRepository.findByUserIdAndIsActiveTrue(id, pageable);
+
+        return offersToOfferRequestWithPaginationDto(pageOfOffer);
+    }
+
+
+    @Override
+    public OfferResponseDto createOffer(OfferCreationDto dto) {
+
+        if (dto == null) throw new OfferCreationDtoIsNullException();
+
+        try {
+            Offer newOffer;
+            try {
```

Comment:
nested try-catch block inside another try-catch block is a bad approach. you can move part of the code into a separate private method that will throw an exception there. or reconsider the logic. 
I propose to discuss this method since it is long and definitely needs refactoring